src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -3424,42 +3432,56 @@ internal static bool TryGetStaticEntitySet(this IEdmPathExpression pathExpressio
-
-                            var nameProp = propertyRef.Properties.FirstOrDefault(e => e.Name == AlternateKeysVocabularyConstants.PropertyRefTypeNamePropertyName);
-                            Debug.Assert(nameProp != null, "expected non null Name Property");
-                            string propertyName = ((IEdmPathExpression)nameProp.Value).PathSegments.FirstOrDefault();
+                        return;
+                    }
 
-                            alternateKey[alias] = type.FindProperty(propertyName);
-                        }
+                    IEdmCollectionExpression keys = ann.Value as IEdmCollectionExpression;
+                    Debug.Assert(keys != null, "expected IEdmCollectionExpression for alternate key annotation value");
 
-                        if (alternateKey.Any())
+                    foreach (IEdmRecordExpression key in keys.Elements.OfType<IEdmRecordExpression>())
+                    {
+                        var edmPropertyConstructor = key.Properties.FirstOrDefault(e => e.Name == "Key");
+                        if (edmPropertyConstructor != null)
                         {
-                            declaredAlternateKeys.Add(alternateKey);
+                            IEdmCollectionExpression collectionExpression = edmPropertyConstructor.Value as IEdmCollectionExpression;
+                            Debug.Assert(collectionExpression != null, "expected IEdmCollectionExpression type for Key Property");
+
+                            IDictionary<string, IEdmProperty> alternateKey = new Dictionary<string, IEdmProperty>();
+                            foreach (IEdmRecordExpression propertyRef in collectionExpression.Elements.OfType<IEdmRecordExpression>())
+                            {
+                                var aliasProp = propertyRef.Properties.FirstOrDefault(e => e.Name == "Alias");
+                                Debug.Assert(aliasProp != null, "expected non null Alias Property");
+                                string alias = ((IEdmStringConstantExpression)aliasProp.Value).Value;
+
+                                var nameProp = propertyRef.Properties.FirstOrDefault(e => e.Name == "Name");
+                                Debug.Assert(nameProp != null, "expected non null Name Property");
+                                string propertyName = ((IEdmPathExpression)nameProp.Value).PathSegments.FirstOrDefault();
+
+                                alternateKey[alias] = type.FindProperty(propertyName);
+                            }
+
+                            if (alternateKey.Any())
+                            {
+                                declaredAlternateKeys.Add(alternateKey);
+                            }
                         }
                     }
-                }
+                };
+
+                // For backwards-compability, we merge the alternate keys from community and core vocabulary annotations.
+
+                retrieveAnnotationAction(annotationValue);

```

Comment:
The output of `GetAlternateKeysAnnotation(this IEdmModel model, IEdmEntityType type)` does not honor the unicity requirements from the [community vocabulary spec](https://github.com/OData/vocabularies/blob/master/OData.Community.Keys.V1.md#216key-declaration-uniqueness):

> Services SHOULD NOT return multiple alternate key definitions for the same entity type that are composed of the exact same set of properties.

The alternate keys are not "merged" as the comment line 3483 calls out, it's instead an union that occurs.